### PR TITLE
Sotfmax bw inference fix

### PR DESCRIPTION
**Pull request by calad0i**
# Description

Fix edge in softmax stable impl norm_t type inference: consider `fixed<2,2> *arr` of range `{-2,-1,0,1}` `max(arr) - arr` can be all `{0,1,2,3}`, so `iwidth` and `width` shall be kept if was signed.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

## Checklist

- [x] all

### CODE REVIEW

**Review by bo3z — APPROVED** [review]

**Comment by JanFSchulte** [comment]
These test failures seem to be reproducible, so I think this requires a further check before merging:

```
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vivado_io_parallel_resource] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vivado_io_parallel_latency] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vivado_io_stream_latency] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vivado_io_stream_resource] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vitis_io_parallel_resource] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vitis_io_parallel_latency] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vitis_io_stream_latency] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
FAILED test_cnn_mnist_qkeras.py::test_accuracy[Vitis_io_stream_resource] - assert (0.988 > 0.92 and 0.011133603238866406 < 0.01)
```
